src/main/java/bio/terra/service/common/azure/StorageTableUtils.java

```diff
@@ -0,0 +1,48 @@
+package bio.terra.service.common.azure;
+
+import java.util.UUID;
+
+public class StorageTableUtils {
+  private static final String DATASET_TABLE_NAME = "dataset";
+  private static final String FILES_TABLE_NAME = "files";
+
+  /**
+   * Generate a Storage Table name from a UUID
+   *
+   * @param resourceId The datasetId or snapshotId to be used as root of the table name
+   * @param suffix suffix for storage table Name
+   * @return A valid azure storage table name
+   */
+  public static String toTableName(String resourceId, StorageTableNameSuffix suffix) {
```

Comment:
I haven't yet read how these are used, but my usual rule of thumb is that if you have a `static` method that also accepts an object, it's likely that the method could be made into a non-static method on that object. In this case I think it could be moved to `StorageTableNameSuffix`. Possibly once that's done the outer class will go away and can be replaced with the enum as a top-level definition.

Reply:
Ok, I've refactored!! Please let me know what you think!

Reply:
Looks good to me!